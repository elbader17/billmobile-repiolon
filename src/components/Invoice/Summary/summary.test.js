import React from 'react';
import { shallow } from 'enzyme';
import SummaryInvoice from './Component';

jest.mock('../../../constants/metrics', () => ({ METRICS: { heightHeader: {} } }));
jest.mock('../../../utils/date', () => ({ presentInvoiceDate: () => {} }));

const fnMock = jest.fn();

const props = {
  fiscalIdentity: {
    category: '',
    identification: '',
    name: ''
  },
  items: [],
  navigation:{navigate: fnMock}
}

test('render correctly component InvoiceSummary', () => {
  const component = shallow(<SummaryInvoice {...props}/>);
  expect(component).toMatchSnapshot();
});

test('call the navigateToInvoice function', () => {
  const component = shallow(<SummaryInvoice {...props}/>);
  const button = component.findWhere(node => node.prop('testID') === 'edit');
  button.simulate('press');
  expect(fnMock).toBeCalled();
});