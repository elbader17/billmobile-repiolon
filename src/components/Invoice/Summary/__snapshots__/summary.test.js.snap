// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render correctly component InvoiceSummary 1`] = `
<ScrollViewMock>
  <View
    style={
      Object {
        "backgroundColor": "#EEEEEE",
        "height": NaN,
        "padding": 18,
      }
    }
  >
    <View
      style={
        Object {
          "backgroundColor": "#fff",
          "borderColor": "#C4C4C4",
          "borderRadius": 4,
          "borderWidth": 1,
        }
      }
    >
      <View
        style={
          Array [
            Object {
              "paddingHorizontal": 12,
              "paddingVertical": 7,
            },
            Object {
              "flexDirection": "row",
              "justifyContent": "space-between",
            },
          ]
        }
      >
        <Text
          style={
            Object {
              "fontFamily": "Lato-Regular",
              "fontSize": 14,
            }
          }
        >
          FACTURA-C
        </Text>
        <Text
          style={
            Object {
              "fontFamily": "Lato-Regular",
              "fontSize": 14,
            }
          }
        />
      </View>
      <View
        style={
          Object {
            "backgroundColor": "#C4C4C4",
            "height": 1.2,
          }
        }
      />
      <View
        style={
          Object {
            "paddingHorizontal": 9,
            "paddingVertical": 7,
          }
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": "#858585",
                "fontFamily": "Lato-Regular",
                "fontSize": 14,
              },
              Object {
                "marginBottom": 5,
              },
            ]
          }
        />
        <Text
          style={
            Object {
              "color": "#C4C4C4",
              "fontFamily": "Lato-Regular",
              "fontSize": 11,
            }
          }
        >
          CUIT: 
        </Text>
        <Text
          style={
            Object {
              "color": "#C4C4C4",
              "fontFamily": "Lato-Regular",
              "fontSize": 11,
            }
          }
        >
          Localidad: 
          Río Cuarto
        </Text>
        <Text
          style={
            Object {
              "color": "#C4C4C4",
              "fontFamily": "Lato-Regular",
              "fontSize": 11,
            }
          }
        >
          Provincia: 
          Córdoba
        </Text>
        <Text
          style={
            Object {
              "color": "#C4C4C4",
              "fontFamily": "Lato-Regular",
              "fontSize": 11,
            }
          }
        >
          IVA: 
          Responsable Inscripto
        </Text>
        <Text
          style={
            Object {
              "color": "#C4C4C4",
              "fontFamily": "Lato-Regular",
              "fontSize": 11,
            }
          }
        >
          Domicilio Fiscal: 
          Dean Funes 645
        </Text>
      </View>
      <View
        style={
          Object {
            "backgroundColor": "#C4C4C4",
            "height": 1.2,
          }
        }
      />
      <View
        style={
          Object {
            "maxHeight": 400,
            "paddingHorizontal": 0,
            "paddingVertical": 7,
          }
        }
      >
        <ScrollViewMock
          style={
            Object {
              "paddingHorizontal": 10,
            }
          }
        />
      </View>
      <View
        style={
          Object {
            "backgroundColor": "#C4C4C4",
            "height": 1.2,
          }
        }
      />
      <View
        style={
          Array [
            Object {
              "paddingHorizontal": 12,
              "paddingVertical": 10,
            },
            Object {
              "flexDirection": "row",
            },
          ]
        }
      >
        <View
          style={
            Object {
              "flex": 2.5,
            }
          }
        >
          <Text
            style={
              Object {
                "color": "#C4C4C4",
                "fontFamily": "Lato-Regular",
                "fontSize": 11,
              }
            }
          >
            CONDICIÓN DE VENTA
          </Text>
          <View
            style={
              Object {
                "backgroundColor": "#EEEEEE",
                "height": 1.2,
                "marginVertical": 5,
              }
            }
          />
          <Text
            style={
              Object {
                "color": "#858585",
                "fontFamily": "Lato-Regular",
                "fontSize": 14,
              }
            }
          >
            CONTADO
          </Text>
        </View>
        <View
          style={
            Array [
              Object {
                "alignItems": "flex-end",
                "flex": 1.7,
                "justifyContent": "center",
              },
              Object {
                "flexDirection": "column",
                "justifyContent": "space-between",
              },
            ]
          }
        >
          <Text
            style={
              Object {
                "color": "#C4C4C4",
                "fontFamily": "Lato-Regular",
                "fontSize": 12,
              }
            }
          >
            SUBTOTAL
          </Text>
          <View
            style={
              Object {
                "borderColor": "#EEEEEE",
                "borderRightWidth": 1.2,
              }
            }
          />
          <Text
            style={
              Object {
                "color": "#C4C4C4",
                "fontFamily": "Lato-Regular",
                "fontSize": 12,
              }
            }
          >
            IMPUESTO
          </Text>
        </View>
        <View
          style={
            Array [
              Object {
                "flex": 2,
              },
              Object {
                "flexDirection": "column",
                "justifyContent": "space-between",
              },
            ]
          }
        >
          <View
            style={
              Object {
                "flexDirection": "row",
                "justifyContent": "space-between",
              }
            }
          >
            <View
              style={
                Array [
                  Object {
                    "borderColor": "#EEEEEE",
                    "borderRightWidth": 1.2,
                  },
                  Object {
                    "marginLeft": 5,
                  },
                ]
              }
            />
            <View>
              <Text
                style={
                  Object {
                    "color": "#858585",
                    "fontFamily": "Lato-Regular",
                    "fontSize": 14,
                  }
                }
              >
                $
              </Text>
            </View>
          </View>
          <View
            style={
              Object {
                "flexDirection": "row",
                "justifyContent": "space-between",
              }
            }
          >
            <View
              style={
                Array [
                  Object {
                    "borderColor": "#EEEEEE",
                    "borderRightWidth": 1.2,
                  },
                  Object {
                    "marginLeft": 5,
                  },
                ]
              }
            />
            <View>
              <Text
                style={
                  Object {
                    "color": "#858585",
                    "fontFamily": "Lato-Regular",
                    "fontSize": 14,
                  }
                }
              >
                $
                0.0
              </Text>
            </View>
          </View>
        </View>
      </View>
      <View
        style={
          Object {
            "backgroundColor": "#C4C4C4",
            "height": 1.2,
          }
        }
      />
      <View
        style={
          Array [
            Object {
              "justifyContent": "center",
              "paddingHorizontal": 12,
              "paddingVertical": 10,
            },
            Object {
              "flexDirection": "row",
              "justifyContent": "space-between",
            },
          ]
        }
      >
        <ForwardRef(Themed.Button)
          buttonStyle={
            Object {
              "backgroundColor": "#fff",
              "borderColor": "#3687d1",
              "borderRadius": 4,
              "borderWidth": 1,
              "height": 66.5,
              "width": 225,
            }
          }
          onPress={[Function]}
          testID="edit"
          title="EDITAR"
          titleStyle={
            Object {
              "color": "#3687d1",
              "fontFamily": "Lato-Regular",
              "fontSize": 14,
            }
          }
        />
        <View
          style={
            Array [
              Object {
                "paddingTop": 4,
                "width": 375,
              },
              Object {
                "flexDirection": "row",
                "justifyContent": "space-between",
              },
            ]
          }
        >
          <Text
            style={
              Object {
                "color": "#858585",
                "fontFamily": "Lato-Regular",
                "fontSize": 17,
              }
            }
          >
            TOTAL
          </Text>
          <Text
            style={
              Object {
                "color": "#858585",
                "fontFamily": "Lato-Regular",
                "fontSize": 18,
              }
            }
          >
            $
          </Text>
        </View>
      </View>
    </View>
    <View
      style={
        Object {
          "bottom": 0,
          "left": 0,
          "position": "absolute",
          "right": 0,
        }
      }
    >
      <ForwardRef(Themed.Button)
        buttonStyle={
          Object {
            "backgroundColor": "#EE6123",
            "borderRadius": 0,
            "height": 106.5,
          }
        }
        disabled={true}
        onPress={[Function]}
        testID="confirmInvoice"
        title="CONFIRMAR FACTURA"
        titleStyle={
          Object {
            "color": "#fff",
            "fontFamily": "Lato-Semibold",
            "fontSize": 14,
          }
        }
      />
    </View>
  </View>
</ScrollViewMock>
`;
